modules/afsocket/afsocket-source.c

```diff
@@ -986,7 +1048,11 @@ afsocket_sd_init_method(LogPipe *s)
         }
     }
 
-  return afsocket_sd_restore_kept_alive_connections(self) && afsocket_sd_open_listener(self);
+  gboolean init_result = afsocket_sd_restore_kept_alive_connections(self) && afsocket_sd_open_listener(self);
+  if (init_result)
+    _afsocket_sd_register_connection_counter(self);
```

Comment:
Why do you register the counter at the end of the init method?

`afsocket_sd_restore_kept_alive_connections()` calls `_connections_count_set`, which requires a valid/registered `num_connections_stats` counter item.

If I'm not mistaken, the counter has to be registered as soon as possible to avoid such issues.